azext_edge/tests/edge/support/create_bundle_int/helpers.py

```diff
@@ -356,40 +380,41 @@ def _get_namespace_determinating_files(name: str, folder: str, file_prefix: str)
         elif _get_namespace_determinating_files(
             name=name, folder=path.join("containerstorage"), file_prefix="configmap"
         ):
+            containerstorage_service = "containerstorage"
             acstor_namespace = name
         elif _get_namespace_determinating_files(
             name=name, folder=OpsServiceType.secretstore.value, file_prefix="deployment"
         ):
             ssc_namespace = name
-        else:
+        elif _get_namespace_determinating_files(name=name, folder=path.join("certmanager"), file_prefix="configmap"):
+            certmanager_namespace = name
+        elif _get_namespace_determinating_files(name=name, folder="meta", file_prefix="instance"):
             namespace = name
 
-    monitor_path = path.join(BASE_ZIP_PATH, arc_namespace, OpsServiceType.azuremonitor.value)
-    for namespace_folder, services in [
-        (clusterconfig_namespace, ["clusterconfig"]),
-        (arc_namespace, ["arcagents"]),
-        (acs_namespace, ["arccontainerstorage"]),
-        (acstor_namespace, ["containerstorage"]),
-        (ssc_namespace, [OpsServiceType.secretstore.value]),
-    ]:
-        if namespace_folder:
-            # remove empty folders in level 1
-            level_1 = walk_result.pop(path.join(BASE_ZIP_PATH, namespace_folder))
+        if _get_namespace_determinating_files(name=name, folder=path.join("certmanager"), file_prefix="certificate"):
+            cert_resource_namespaces.append(name)
 
-            if namespace_folder == arc_namespace and monitor_path in walk_result:
-                services.append(OpsServiceType.azuremonitor.value)
-            assert set(level_1["folders"]) == set(services)
-            assert not level_1["files"]
+    # find the acstor namespace if fault tolerance is not enabled
+    if not acstor_namespace:
+        # acstor_namespace should be the namespace that is not in the list of namespaces that are not certmanager
+        acstor_namespace = set(cert_resource_namespaces) - set([certmanager_namespace, arc_namespace, namespace])
```

Comment:
a bit confused here - also pretty sure there should be a way to pop without failing if the item is not there

Reply:
this is used when we need to find the acstor namespace when fault tolerance is not enabled, since namespace like cert-manager, acstor, arc and aio will always have cert manager resources, therefore to find the acstor, we just need to get all namespaces that contains cert manager resources, and then remove the already found cert-manager, arc and aio namespaces

Reply:
updated the comments